src/main/start-main-application/lens-window/application-window/create-electron-window.injectable.ts

```diff
@@ -79,6 +87,14 @@ const createElectronWindowInjectable = getInjectable({
         },
       });
 
+      if (isLinux) {
+        const iconFileName = getAbsolutePath(lensResourcesDir, "../open-lens.png");
+
+        if (existsSync(iconFileName)) {
+          browserWindow.setIcon(iconFileName);
+        }
```

Comment:
Don't know if this with pass CI...

In any case I think using `pathExistsInjectable` and then having this be

```ts
pathExists(iconFileName)
  .then(iconExists => {
    if (iconExists) {
      browserWindow.setIcon(iconFileName);
    }
  });
```

Reply:
I tried async/promise but in the (unlikely) eventuality of nag screen going away before promise resolution, it might generate an error message but I can encolse it in a try / catch

Reply:
Code is going to be

```node
      if (isLinux) {
        const iconFileName = getAbsolutePath(lensResourcesDir, "../open-lens.png");

        pathExists(iconFileName)
          .then(iconExists => {
            if (iconExists) {
              try {
                browserWindow.setIcon(iconFileName);
              } catch (err) {
                logger.warn(`Error while setting window icon ${err}`);
              }
            }
          });
      }
```

Reply:
This part is working. Remains the icon location mystery :)